python/pyspark/errors/utils.py

```diff
@@ -119,3 +124,59 @@ def get_message_template(self, error_class: str) -> str:
             message_template = main_message_template + " " + sub_message_template
 
         return message_template
+
+
+def _capture_call_site(fragment: str) -> None:
+    """
+    Capture the call site information including file name, line number, and function name.
+    This function updates the thread-local storage from server side (PySparkCurrentOrigin)
```

Comment:
> server side

sorry, probably I don't have enough context. Does this implementation also support spark connect?

Reply:
I wanted to mention JVM side. Let me clarify it